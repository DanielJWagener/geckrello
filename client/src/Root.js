// Simply creates a Redux store and passes it in to a provider tag.
// This logic was originally in index.js, but we've movied it here to
// make our testing process easier.
import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "./reducers";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducers,
  {},
  composeEnhancers(applyMiddleware(reduxThunk))
);

export default ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
