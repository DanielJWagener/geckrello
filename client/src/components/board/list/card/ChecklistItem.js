import React from "react";
import { connect } from "react-redux";
import { checklistCheck, checklistDelete } from "../../../../actions";

function ChecklistItem(props) {
  function checkItem() {
    props.checklistCheck(props.cardId, props.checklistItemId);
  }

  function deleteItem() {
    props.checklistDelete(props.cardId, props.checklistItemId);
    props.signalDelete();
  }

  if (!props.checked) {
    return (
      <div className="item">
        <div
          className="item__checkbox item__checkbox--unchecked"
          onClick={checkItem}
        ></div>
        <div className="item__title">{props.checklistItemTitle}</div>
        <div className="item__delete" onClick={deleteItem}>
          &times;
        </div>
      </div>
    );
  } else {
    return (
      <div className="item">
        <div
          className="item__checkbox item__checkbox--checked"
          onClick={checkItem}
        >
          <div className="checkmark">&#10004;</div>
        </div>
        <div className="item__title">{props.checklistItemTitle}</div>
        <div className="item__delete" onClick={deleteItem}>
          &times;
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  const { checklistItemId, cardId } = ownProps;
  let card = state.cards.filter(card => card.cardId === cardId)[0];
  if (card.checklist.length > 0) {
    let checked = card.checklist.filter(
      item => item.checklistItemId === checklistItemId
    )[0].checked;

    return { checked };
  }
};

export default connect(
  mapStateToProps,
  { checklistCheck, checklistDelete }
)(ChecklistItem);
