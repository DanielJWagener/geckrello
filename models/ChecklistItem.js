const mongoose = require("mongoose");
const { Schema } = mongoose;

const checklistItemSchema = new Schema({
  label: String,
  checked: { type: Boolean, default: false }
});

module.exports = checklistItemSchema;
