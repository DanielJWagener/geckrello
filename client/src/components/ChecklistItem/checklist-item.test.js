import { shallow } from "enzyme";
import React from "react";
import ChecklistItem from "./checklist-item.component";
import { ChecklistItem as ChecklistItemPure } from "./checklist-item.component";
import Root from "../../Root";

it("renders ChecklistItem component", () => {
  expect(shallow(<ChecklistItemPure />).debug()).toMatchSnapshot();
});
