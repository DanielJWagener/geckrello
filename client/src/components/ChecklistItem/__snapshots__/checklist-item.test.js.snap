// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ChecklistItem component 1`] = `
"<div className=\\"item\\">
  <div className=\\"item__checkbox item__checkbox--unchecked\\" onClick={[Function: checkOrUncheck]} />
  <div className=\\"item__title\\" />
  <div className=\\"item__delete\\" onClick={[Function: deleteItem]}>
    Ã—
  </div>
</div>"
`;
